micrometer-core/src/test/java/io/micrometer/core/instrument/binder/jvm/JvmGcMetricsTest.java

```diff
@@ -65,13 +64,18 @@ void noJvmImplementationSpecificApiSignatures() {
     }
 
     @Test
-    void metersAreBound() {
-        assertThat(registry.find("jvm.gc.live.data.size").gauge()).isNotNull();
-        assertThat(registry.find("jvm.gc.memory.allocated").counter()).isNotNull();
-        assertThat(registry.find("jvm.gc.max.data.size").gauge().value()).isGreaterThan(0);
+    void gcMetricsAvailableAfterGc() {
+        System.gc();
+        await().timeout(200, TimeUnit.MILLISECONDS).alias("NotificationListener takes time after GC")
+                .untilAsserted(() ->
+                        assertThat(registry.find("jvm.gc.live.data.size").gauge().value()).isPositive());
+        assertThat(registry.find("jvm.gc.memory.allocated").counter().count()).isPositive();
+        assertThat(registry.find("jvm.gc.max.data.size").gauge().value()).isPositive();
 
-        assumeTrue(binder.isGenerationalGc);
```

Comment:
When the condition failed, the whole test was marked as skipped, but if we want to use this same test for generational and non-generational collectors, we should run the part of this test before here for both and report it as passed (not skipped) for non-generational collectors if all assertions passed before here.